src/libs/actions/PersonalDetails.js

```diff
@@ -186,6 +200,36 @@ function getFromReportParticipants(reports) {
         });
 }
 
+
+/**
+ * Sets the personal details object for the current user
+ *
+ * @param {Object} details
+ */
+function setPersonalDetails(details) {
+    API.PersonalDetails_Update({details});
+    NameValuePair.set(CONST.NVP.TIMEZONE, details.timezone);
+
+    // Update the associated onyx keys
+    Onyx.merge(ONYXKEYS.MY_PERSONAL_DETAILS, details);
+    Onyx.merge(ONYXKEYS.PERSONAL_DETAILS, formatPersonalDetails({[currentUserEmail]: details}));
+}
+
+/**
+ * Sets the user's avatar image
+ *
+ * @param {String} base64image
+ */
+function setAvatar(base64image) {
+    API.User_UploadAvatar({base64image}).then((response) => {
+        // Once we get the s3url back, update the personal details for the user with the new avatar URL
+        if (response.jsonCode === 200) {
+            myPersonalDetails.avatar = response.s3url;
+            setPersonalDetails(myPersonalDetails);
```

Comment:
I didn't realize this before, but it seems maybe incorrect to modify the `myPersonalDetails` directly. 

Typically we would modify the key in Onyx and then it would just update in the callback that way the local variables in the "connect" callbacks are matching what is in local storage.

I _think_ this API can handle partial details so we could just send `{avatar: response.s3url}` and also do something like 
```js
Onyx.merge(ONYXKEYS.MY_PERSONAL_DETAILS, {avatar: response.s3url});
```

Reply:
Nice, updated this, retested, and it works as expected!